packages/react-forms/src/InputDate/__mocks__/InputDateWrapper.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) Quid, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+// @flow
+import React, { useState, useCallback, Fragment } from 'react';
+import InputDate from '../index';
+
+type InputDateWrapperProps = {
+  defaultValue: string,
+  defaultCalendarValue?: Date,
+  onChange?: string => void,
+  onCalendarChange?: Date => void,
+  isOpen?: boolean,
+};
+
+const InputDateWrapper = ({
+  onChange,
+  defaultValue,
+  onCalendarChange,
+  defaultCalendarValue,
+  isOpen: isOpenProp,
+  ...props
+}: InputDateWrapperProps) => {
+  const [isOpen, setOpen] = useState(isOpenProp);
+  const [value, setValue] = useState(defaultValue);
+  const [calendarValue, setCalendarValue] = useState(
```

Comment:
We should use the @quid/recat-use-controlled-state to make sure the user can switch between controlled and uncontrolled while the component is already instantiated. It should also simplify the code

Reply:
Is there any reason you commented this in the test?

Reply:
It was so long I thought it was the actual implementation 😂